get-cabal-flags.hs

```diff
@@ -0,0 +1,8 @@
+import Data.Version (Version (Version))
+import Distribution.Simple.Utils (cabalVersion)
+
+main :: IO ()
+main =
+  putStrLn $ if cabalVersion >= Version [1,22] []
+             then "(\"-DuseCompilerInfo\")"
+             else "()"
```

Comment:
@lunaryorn I will admit that I find the haskell code more intuitive than emacs-subprocess stuff, but I think you're right, it's probably the better direction.  I'll see what I can do.